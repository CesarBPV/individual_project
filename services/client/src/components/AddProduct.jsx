import React from 'react';
const AddProduct= (props) => {
    return (
    <form onSubmit={(event) => props.addProduct(event)}>
        <div className="field">
        <input
        name="name"
        className="input is-large"
        type="text"
        placeholder="Enter a name"
        required
        value={props.name}
        onChange={props.handleChange}
        />
        </div>
        <div className="field">
        <input
        name="stock"
        className="input is-large"
        type="text"
        placeholder="Enter stock"
        required
        value={props.stock}
        onChange={props.handleChange}
        />
        </div>
        <div className="field">
        <input
        name="price"
        className="input is-large"
        type="text"
        placeholder="Enter price"
        required
        value={props.price}
        onChange={props.handleChange}
        />
        </div>
        <div className="field">
        <input
        name="trademark"
        className="input is-large"
        type="text"
        placeholder="Enter trademark"
        required
        value={props.trademark}
        onChange={props.handleChange}
        />
        </div>
        <div className="field">
        <input
        name="category"

        className="input is-large"
        type="text"
        placeholder="Enter category"
        required
        value={props.category}
        onChange={props.handleChange}
        />
        </div>
        <input
        type="submit"
        className="button is-primary is-large is-fullwidth"
        value="Submit"
        />
        </form>
        )
    };

export default AddProduct;