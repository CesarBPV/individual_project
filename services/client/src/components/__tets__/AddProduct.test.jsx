import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import AddProduct from '../AddProduct';

test('AddProduct renders properly', () => {
    const wrapper = shallow(<AddProduct/>);
    const element = wrapper.find('form');
    expect(element.find('input').length).toBe(6);
    expect(element.find('input').get(0).props.name).toBe('name');
    expect(element.find('input').get(1).props.name).toBe('stock');
    expect(element.find('input').get(2).props.name).toBe('price');
    expect(element.find('input').get(3).props.name).toBe('trademark');
    expect(element.find('input').get(4).props.name).toBe('category');
    expect(element.find('input').get(5).props.type).toBe('submit');
});

test('AddProduct renders a snapshot properly', () => {
    const tree = renderer.create(<AddProduct/>).toJSON();
    expect(tree).toMatchSnapshot();
});
