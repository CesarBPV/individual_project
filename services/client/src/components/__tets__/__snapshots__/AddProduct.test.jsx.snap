// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProduct renders a snapshot properly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="name"
      placeholder="Enter a name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="stock"
      placeholder="Enter stock"
      required={true}
      type="text"
    />
  </div>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="price"
      placeholder="Enter price"
      required={true}
      type="text"
    />
  </div>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="trademark"
      placeholder="Enter trademark"
      required={true}
      type="text"
    />
  </div>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="category"
      placeholder="Enter category"
      required={true}
      type="text"
    />
  </div>
  <input
    className="button is-primary is-large is-fullwidth"
    type="submit"
    value="Submit"
  />
</form>
`;
